auction_keeper/main.py

```diff
@@ -318,6 +320,16 @@ def approve(self):
         if self.collateral:
             self.collateral.approve(self.our_address, gas_price=self.gas_price)
 
+    def plunge(self):
+        pending_txes = get_pending_transactions(self.web3)
+        if len(pending_txes) > 0:
+            while len(pending_txes) > 0:
+                logging.warning(f"Cancelling first of {len(pending_txes)} pending transactions")
+                pending_txes[0].cancel(gas_price=self.gas_price)
+                # After the synchronous cancel, wait to see if subsequent transactions get mined
+                time.sleep(28)
```

Comment:
Is 28 seconds enough time? 

Fast price with `ethgasstation` is expected to be [mined within two minutes](https://docs.ethgasstation.info/gas-price), so IIUC, if the `gas-initial-multiplier` is 1, there's a possibility this doesn't wait long enough for a pending transaction to be overridden.

Reply:
If the TX queue is stuck, later transactions would likely have already been replaced up to a large max gas limit (the default is 2000 GWEI).  Waiting two blocks should give these high-gas TXes a chance to be mined before pushing a cancellation TX with the same nonce onto the mempool.

Reply:
And if later transactions _didn't_ have high gas, we'd rather cancel than wait (seemingly) infinite time to free the TX queue.